File: src/exercicios/04_Subalgoritmos/01-int_positivo/int_positivo.py
#  -*- coding: utf-8 -*-
#       @file: int_positivo.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Implemente a função leia_inteiro_positivo.


def leia_inteiro():
    '''Retorna um valor inteiro lido da entrada padrão.'''

    return int(input('Digite o número inteiro: '))


def leia_inteiro_positivo():
    '''Retorna um valor inteiro positivo lido da entrada padrão.

    Lê valor(es) da entrada padrão até receber um valor inteiro positivo, que é
    retornado.
    '''

    # Use a função leia_inteiro()


N = input()

for x in xrange(N):
    num = leia_inteiro_positivo()
    print 'Li', num

File: src/exercicios/04_Subalgoritmos/01-int_positivo/int_positivo.c
/**      @file: int_positivo.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Implemente a função leia_inteiro_positivo. */

#include <stdio.h>

/* Retorna um valor inteiro lido da entrada padrão. */
int leia_inteiro() {
    int num;
    printf("Digite o número inteiro: ");
    scanf("%d", &num);
    return num;
}

/* Lê valor(es) da entrada padrão até receber um valor
 * inteiro positivo, que é retornado. */
int leia_inteiro_positivo() {
    /* Use a função leia_inteiro(). */
}

int main() {
    int N;

    scanf("%d", &N);

    while(N--)
        printf("Li %d\n", leia_inteiro_positivo());

    return 0;
}